### ISSUE TITLE
Ability to remove event listener from even bus

### ISSUE THREAD

**Issue by codenamestif** [atomashpolskiy/bt#45]
It looks like since you added event listener to the even bus, you can't remove it. It would be nice to have such ability to remove listeners by reference or by some id.

**Comment by codenamestif** [comment]
Or i guess it could be event better if event bus has client-specific scope i.e each client has its own event bus and listeners. So once client is done, so all the client-related things will be collected by GC.

**Comment by atomashpolskiy** [comment]
I'd prefer to stick with one event bus per runtime, but this is definitely a (minor) memory leak, so it might make sense to remove torrent-specific listeners, once the torrent is done or cancelled (i.e. the corresponding BtClient has terminated). The best place to do this should be somewhere in the processing chain (see bt.processor.TorrentProcessorFactory.)

**Comment by ckovorodkin** [comment]
> but this is definitely a (minor) memory leak

Each successful handshake adds two listeners. This can be a serious memory leak. Especially when remote peer is malignant and creates incoming connections in loop.

https://github.com/atomashpolskiy/bt/blob/6a17e0ed94fa6ff7b2362812f8421c65a0fa3132/bt-core/src/main/java/bt/net/PeerConnectionFactory.java#L191-L202

**Comment by atomashpolskiy** [comment]
I guess it's around 30 bytes per connection, does not seem so critical. But again, I agree that this needs to be addressed.
